src/jit/optimizer.cpp

```diff
@@ -6142,9 +6142,15 @@ bool Compiler::optHoistLoopExprsForTree(GenTreePtr        tree,
         childrenCctorDependent[i] = false;
     }
 
-    // Initclass CLS_VARs are the base case of cctor dependent trees.
-    bool treeIsCctorDependent = (tree->OperIs(GT_CLS_VAR) && ((tree->gtFlags & GTF_CLS_VAR_INITCLASS) != 0));
-    bool treeIsInvariant      = true;
+    // Initclass CLS_VARs and IconHandles are the base cases of cctor dependent trees.
+    // In the IconHandle case, it's of course the dereference, rather than the constant itself, that is
+    // truly dependent on the cctor.  So a more precise approach would be to separately propagate
+    // isCctorDependent and isAddressWhoseDereferenceWouldBeCctorDependent, but we don't for simplicity/throughput;
+    // the constant itself would be considered non-hoistable anyway, since optIsCSEcandidate returns
+    // false for constants.
```

Comment:
`since optIsCSEcandidate returns false for constants.`

This might not continue to be true in the future.
Is it possible to flag the indirection instead of the constant?
Or are we out of flag bits on GT_IND?